packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js

```diff
@@ -77,6 +107,22 @@ export default class {
       .listObjects(this.projectId, containerId)
       .then((container) => {
         this.containerObjects = container.objects;
+        // try to detect correct job type and file
+        const pythonFiles = container.objects.filter((object) =>
+          PYTHON_FILES_CONTENT_TYPES.includes(object.contentType),
+        );
+        const javaFiles = container.objects.filter((object) =>
+          JAVA_FILES_CONTENT_TYPES.includes(object.contentType),
+        );
+        if (pythonFiles.length > 0) {
+          this.state.jobType = JOB_TYPE_PYTHON;
+          this.state.mainApplicationCode = pythonFiles[0].name;
+        } else if (javaFiles.length > 0) {
+          this.state.jobType = JOB_TYPE_JAVA;
+          this.state.mainApplicationCode = javaFiles[0].name;
```

Comment:
**question**: what if the java files have different `name` value ?

Reply:
Same as here: https://github.com/ovh/manager/pull/7069#discussion_r926408399